src/EFCore/Metadata/Builders/EntityTypeBuilder.cs

```diff
@@ -575,23 +575,35 @@ private OwnedNavigationBuilder OwnsManyBuilder(in TypeIdentity ownedType, string
         ///     no property is specified, the relationship will be configured without a navigation property on this
         ///     end.
         /// </param>
+        /// <param name="navigationConfiguration">
+        ///     An optional action which further configures the navigation property.
+        /// </param>
         /// <returns> An object that can be used to configure the relationship. </returns>
         public virtual ReferenceNavigationBuilder HasOne(
             [NotNull] string relatedTypeName,
-            [CanBeNull] string navigationName)
+            [CanBeNull] string navigationName,
+            [CanBeNull] Action<NavigationBuilder> navigationConfiguration = null)
         {
             Check.NotEmpty(relatedTypeName, nameof(relatedTypeName));
             Check.NullButNotEmpty(navigationName, nameof(navigationName));
 
             var relatedEntityType = FindRelatedEntityType(relatedTypeName, navigationName);
+            var foreignKey = Builder.HasRelationship(
+                    relatedEntityType, navigationName, ConfigurationSource.Explicit,
+                    targetIsPrincipal: Builder.Metadata == relatedEntityType ? true : (bool?)null).Metadata;
+
+            if (navigationConfiguration != null
+                && foreignKey?.DependentToPrincipal != null)
+            {
```

Comment:
You can't assume that the navigation is on the dependent, since this might be a 1:1 relationship. Use this to get the navigation:
```suggestion
            if (navigationConfiguration != null
                && navigationName != null)
            {
                var navigation = Builder.Metadata == relatedEntityType || foreignKey.PrincipalEntityType == relatedEntityType
                        ? foreignKey.DependentToPrincipal
                        : foreignKey.PrincipalToDependent;
```

Reply:
That works. But I have 1:1 test cases both ways round. What test case am I missing?

Reply:
```C#
                modelBuilder.Entity<NavDependent>()
                    .HasOne()
                    .WithOne(
                        e => e.Dependent,
                        nb => nb.UsePropertyAccessMode(PropertyAccessMode.Field))
                    .HasForeignKey<NavDependent>();
                    
                modelBuilder.Entity<OneToOneNavPrincipal>()
                    .HasOne(
                        e => e.Dependent,
                        nb => nb.UsePropertyAccessMode(PropertyAccessMode.Property))
                    .WithOne();
                    
                var principal = (IEntityType)model.FindEntityType(typeof(OneToOneNavPrincipal));

                Assert.Equal(PropertyAccessMode.Property, principal.FindNavigation("Dependent").GetPropertyAccessMode());
```

Also add some update and query tests  to make sure the field is used by having the entity types look like:

```C#
        private class OneToOneFieldNavPrincipal
        {
            public int Id { get; set; }
            public string Name { get; set; }
            
            public NavDependent _dependent;

            public NavDependent Dependent
            {
                get => throw new NotImplementedException();
                set => throw new NotImplementedException();
            }
        }
```

Reply:
I added the first test. Working on the 2nd.

Reply:
Added the 2nd test.